sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -341,6 +347,33 @@ class FileStreamSourceSuite extends FileStreamSourceTest with SharedSQLContext {
     Utils.deleteRecursively(tmp)
   }
 
+  test("metadataPath should be in checkpointLocation") {
```

Comment:
`metadataPath` is only for `FileStreamSource` so I think this test belongs to `FileStreamSourceSuite`.

I added a test to test source ids in `DataFrameReaderWriterSuite`.